lib/utils/tls.go

```diff
@@ -35,33 +35,22 @@ import (
 // ListenTLS sets up TLS listener for the http handler, starts listening
 // on a TCP socket and returns the socket which is ready to be used
 // for http.Serve
-func ListenTLS(address string, certFile, keyFile string) (net.Listener, error) {
-	tlsConfig, err := CreateTLSConfiguration(certFile, keyFile)
+func ListenTLS(address string, certFile, keyFile string, cipherSuites []uint16) (net.Listener, error) {
+	tlsConfig, err := CreateTLSConfiguration(certFile, keyFile, cipherSuites)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
 	return tls.Listen("tcp", address, tlsConfig)
 }
 
 // TLSConfig returns default TLS configuration strong defaults.
-func TLSConfig() *tls.Config {
+func TLSConfig(cipherSuites []uint16) *tls.Config {
 	config := &tls.Config{}
 
-	// Only support modern ciphers (Chacha20 and AES GCM). Key exchanges which
-	// support perfect forward secrecy (ECDHE) have priority over those that do
-	// not (RSA).
-	config.CipherSuites = []uint16{
-		tls.TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305,
-		tls.TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305,
-
-		tls.TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,
-		tls.TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,
-
-		tls.TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,
-		tls.TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,
-
-		tls.TLS_RSA_WITH_AES_128_GCM_SHA256,
-		tls.TLS_RSA_WITH_AES_256_GCM_SHA384,
+	// If ciphers suites were passed in, use them. Otherwise use the the
+	// Go defaults.
```

Comment:
why not use our own defaults (that you deleted) here? I think they are more restrictive than Go's defaults

Reply:
That's what happens for code running within Teleport, but instead we pass in the cipher suites to allow users to configure what cipher suites they want to support.

However, for clients like tsh where we don't have a configuration file, we need to support a very permissive set of cipher suites. For example, say a user starts Teleport with the following configuration:

```
teleport
  ciphersuites:
    - "tls-rsa-with-aes-128-cbc-sha"
    - "tls-rsa-with-aes-256-cbc-sha"
    - "tls-rsa-with-aes-128-cbc-sha256"
```

If we use Teleport defaults for clients like `tsh`, it will never be able to connect to Teleport because our defaults only support `AES-GCM` and `chacha20`.



We do in most places, that's what's passed in to this function. However there are a few places